Frameworks/CoreText/CTFontManager.mm

```diff
@@ -34,53 +34,103 @@ CFArrayRef CTFontManagerCreateFontDescriptorsFromURL(CFURLRef fileURL) {
     return StubReturn();
 }
 
+// Helper functions to reduce code duplication, as (un)registering fonts have almost identical path
+// (Un)Registers font by creating temporary CFArray and calling into function for multiple fonts
+template <typename TLambda>
+static bool __CTFontManagerUpdateWithFont(CFURLRef fontURL, CTFontManagerScope scope, CFErrorRef _Nullable* error, const TLambda&& func) {
+    woc::unique_cf<CFArrayRef> fontURLs{ CFArrayCreate(nullptr, (const void**)&fontURL, 1, &kCFTypeArrayCallBacks) };
+    if (error) {
+        CFArrayRef errors = nil;
+        bool ret = func(fontURLs.get(), scope, &errors);
+        if (errors != nil) {
+            *error = (CFErrorRef)CFRetain(CFArrayGetValueAtIndex(errors, 0));
+        }
+
+        CFRelease(errors);
+        return ret;
+    }
+
+    return func(fontURLs.get(), scope, nullptr);
+}
+
+// Converts CFURLs to CFDatas which are passed into DWriteWrapper methods
+template <typename TLambda>
```

Comment:
>TLambda [](start = 19, length = 7)

Please add a comment documenting the typical parameter set and return type expected of a lambda